AP_HAL_Chibios hwdef script: add h757 all ADC2 and ADC3 pins

magicrub opened the pull request:
This implements the rest of the ADC2 and ADC3 pins in the H757 build script. However, some pins are named ending in "_C" which breaks the parser because it's expecting the pin to only have a number. This currently only effects one pin on CubeRedPrimary. They're using the PC3_C pin but labeling it as PC3 but those are two pysical different pins.

There's only a couple of these pins so we get around it. here's the three pins it effects:
![h757pinspng](https://github.com/ArduPilot/ardupilot/assets/4782875/ab1e6560-d593-4f49-8ed9-91ca72b284a7)

MallikarjunSE commented:
Hey @magicrub 
Can this be added to H743 as well? I have ADC on PC3_C too for 5V sens. 
Thank you!

tridge commented on libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32H757xx.py:
```diff
@@ -1278,40 +1278,72 @@
-    "PA7"	:	7,
-    "PC5"	:	8,
-    "PB0"	:	9,
+	"PA0_C"	:	0,
```
I would do it without the _C, we normalise other pin names in these files as well, it varies based on which person in ST creates the particular datasheet

magicrub replied:
The _C is the whole problem. Did you see the image in the description?

magicrub replied:
The easiest thing to do is just remove support for those pins on the 240+25 pin count versions

magicrub commented:
I removed support for the conflicting _C pins which only effects the the TFBGA240+25 14x14mm chip. However, it seems several hwdef's use PC2 on ADC1 which, unless it's the giant pincount BGA chip, it shouldn't be working. Can anyone confirm that on a BeastH7 or ARKV6XI? I also updated H743's ADCs because it appears they match H757 exactly... so, I got you @MallikarjunSE !

MallikarjunSE commented:
> I removed support for the conflicting _C pins which only effects the the TFBGA240+25 14x14mm chip. However, it seems several hwdef's use PC2 on ADC1 which, unless it's the giant pincount BGA chip, it shouldn't be working. Can anyone confirm that on a BeastH7 or ARKV6XI? I also updated H743's ADCs because it appears they match H757 exactly... so, I got you @MallikarjunSE ! 

Thank you so much!

magicrub commented:
@bugobliterator any thoughts on this? Care to verify?
